### DISCUSSION TITLE
SQL Query, GraphQL where to find a valid schema definition for an array of string?

### DISCUSSION THREAD

**Discussion by MarGraz** [OrchardCMS/OrchardCore#15992] [Q&A]
Hi everyone,

I have a SQL query that returns several fields, including one that contains an array of strings. **I want to return this via GraphQL, so I need to define a schema**, but I'm not sure where to find a valid schema definition.

This is the **object returned by the query**:

```json
[
    {
        "DocumentId": 471,
        "userid": "user789",
        "managerid": "ah342",
        "isactive": "true",
        "name": "Pippo Pluto",
        "creationdate": "2024-05-03T00:00:00Z",
        "distributorvalues": [
            "\u00221\u0022",
            "\u00222\u0022"
        ]
     },
    ...
]
```
   
**I tried adding this schema**, but it doesn't work:

```json
{
    "type": "object",
    "properties": {
        "DocumentId": {
            "type": "integer"
        },
        "userid": {
            "type": "string"
        },
        "managerid": {
            "type": "string"
        },
        "isactive": {
            "type": "string",
            "enum": ["true", "false"]
        },
        "name": {
            "type": "string"
        },
        "creationdate": {
            "type": "string",
            "format": "date-time"
        },
        "distributorvalues": {
            "type": "string"
        }
    },
}
```

I also tried to follow [these guidelines](https://graphql.org/learn/schema/), but without success. Same problem with the [official documentation](https://docs.orchardcore.net/en/latest/reference/core/Apis.GraphQL.Abstractions/#define-a-query-type).

1. Where can I find the **GraphQL schema definition for Orchard Core** with all the supported types?
2. Is there a way to **pass filter parameters**, such as a Date or ID, to an SQL query through GraphQL? Or I need to define a custom GraphQL schema?

Thank you

**Comment by Piedone** [comment]
I won't be able to answer this (without digging into it a lot) since I've never done it, but can you clarify your end goal? What are you building? Maybe there's a simpler approach.

**Reply by MarGraz** [reply]
I'm working on several Orchard Core microservices to manage a webshop that also needs to communicate with a custom CRM. 
**Initially, I planned to handle all Content Items GET requests using GraphQL** and apply filters such as `Date` and my custom text field `UserId`. However, **this approach seems tricky** without implementing n models and extending the GraphQL schema.

**In the current discussion**, I attempted to GET one of my Content Items and display the property `distributorvalues` as an array, because this property contains an array of strings. However, without knowing the GraphQL schema accepted by Orchard Core, this task is challenging.

I think **I will consider using an `ApiController` for all CRUD actions**, since I'm spending too much time achieving little progress.

Thank you

**Reply by Piedone** [reply]
OK, thanks!
